pom.xml

```diff
@@ -259,9 +259,37 @@
                           <environmentVariables>
                               <AWS_PROFILE>${AWS_PROFILE}</AWS_PROFILE>
                               <AWS_REGION>${AWS_REGION}</AWS_REGION>
-                              <S3_BUCKET>${S3_BUCKET}</S3_BUCKET>
-                              <S3_DIR>${S3_DIR}</S3_DIR>
                           </environmentVariables>
+                          <properties>
+                              <io.jenkins.plugins.artifact_manager_s3.S3BlobStore.container>${S3_BUCKET}</io.jenkins.plugins.artifact_manager_s3.S3BlobStore.container>
+                              <io.jenkins.plugins.artifact_manager_s3.S3BlobStore.prefix>${S3_DIR}</io.jenkins.plugins.artifact_manager_s3.S3BlobStore.prefix>
+                          </properties>
+                      </configuration>
+                  </plugin>
+              </plugins>
+          </build>
+      </profile>
+      <profile>
+          <id>S3-settings-properties</id>
+          <activation>
+              <property>
+                  <name>io.jenkins.plugins.artifact_manager_s3.S3BlobStore.container</name>
+              </property>
+          </activation>
+          <build>
+              <plugins>
+                  <plugin>
+                      <groupId>org.apache.maven.plugins</groupId>
+                      <artifactId>maven-surefire-plugin</artifactId>
+                      <configuration>
+                          <environmentVariables>
+                              <AWS_PROFILE>${AWS_PROFILE}</AWS_PROFILE>
+                              <AWS_REGION>${AWS_REGION}</AWS_REGION>
```

Comment:
Wait, we still have environment variables? This seems confusing no?

Reply:
those are AWS related we can not touch them.

Reply:
Sorry, why? It seems that if we are switching to UI-driven configuration, everything should be configurable in a consistent manner. What is special about the profile & region? Maybe this is tied into explicit selection of credentials?

Reply:
yep, maybe it is more for a local test when you have to generate the token with aws CLI, maybe we can remove then.

Reply:
Yeah somehow we need to be able to pass in relevant bits of configuration to _tests_ either via environment variables or a local (`~/.m2/settings.xml`) profile, but we can pick anything that is convenient and works for both local development and CI. Initially I had defined this shared profile which worked in conjunction with a local profile, but then I actually stopped using it (in favor of setting environment variables prior to running `mvn test`) and just did not bother to delete it.